packages/xrpc-server/src/util.ts

```diff
@@ -81,93 +79,88 @@ export function decodeQueryParam(
   }
 }
 
-export function getQueryParams(url = ''): Record<string, string | string[]> {
-  const { searchParams } = new URL(url ?? '', 'http://x')
-  const result: Record<string, string | string[]> = {}
+export type QueryParams = Record<string, undefined | string | string[]>
+export function getQueryParams(url = ''): QueryParams {
+  const result: QueryParams = Object.create(null)
+
+  const queryStringIdx = url.indexOf('?')
+  if (queryStringIdx === -1) return result
+
+  const queryString = url.slice(queryStringIdx + 1)
+  if (queryString === '') return result
+
+  const searchParams = new URLSearchParams(queryString)
   for (const key of searchParams.keys()) {
-    result[key] = searchParams.getAll(key)
-    if (result[key].length === 1) {
-      result[key] = result[key][0]
-    }
+    const values = searchParams.getAll(key)
+    result[key] = values.length === 1 ? values[0] : values
   }
   return result
 }
 
-export function validateInput(
+export function createInputVerifier(
```

Comment:
If I am understanding correctly there are no changes of behavior intended in here.  I am pretty sure that is the case, but could you help confirm?

I think we gained a lot of clarity in the implementation here over all 👍